platform/mellanox/mlnx-platform-api/sonic_platform/eeprom.py

```diff
@@ -47,16 +49,19 @@ class Eeprom(eeprom_tlvinfo.TlvInfoDecoder):
     EEPROM_DECODE_CONTENT = 2
 
     def __init__(self):
-        for attempt in range(self.RETRIES):
-            if not os.path.islink(EEPROM_SYMLINK):
-                time.sleep(1)
-            else:
-                break  
-
-        if not (os.path.exists(EEPROM_SYMLINK) \
-                or os.path.isfile(os.path.join(CACHE_ROOT, CACHE_FILE))):
-            log_error("Nowhere to read syseeprom from! No symlink or cache file found")
-            raise RuntimeError("No syseeprom symlink or cache file found")
+        self.machine_info = get_machine_info()
+        self.onie_platform = self.machine_info['onie_platform']
+        if 'simx' not in self.onie_platform:
+            for attempt in range(self.RETRIES):
+                if not os.path.islink(EEPROM_SYMLINK):
+                    time.sleep(1)
+                else:
+                    break  
+    
+            if not (os.path.exists(EEPROM_SYMLINK) \
+                    or os.path.isfile(os.path.join(CACHE_ROOT, CACHE_FILE))):
+                log_error("Nowhere to read syseeprom from! No symlink or cache file found")
+                raise RuntimeError("No syseeprom symlink or cache file found")
```

Comment:
@tomer-israel suggest to move this code to the separate method: this will improve readability:
```python
            for attempt in range(self.RETRIES):
                if not os.path.islink(EEPROM_SYMLINK):
                    time.sleep(1)
                else:
                    break  

            if not (os.path.exists(EEPROM_SYMLINK) \
                    or os.path.isfile(os.path.join(CACHE_ROOT, CACHE_FILE))):
                log_error("Nowhere to read syseeprom from! No symlink or cache file found")
                raise RuntimeError("No syseeprom symlink or cache file found")
```